solmanager.py

```diff
@@ -1227,15 +1241,27 @@ def _exec_cmd(self,cmd):
         return returnVal
 
 class SolManager(object):
-    
-    def __init__(self,serialport,tcpport):
+
+    def __init__(self):
         AppData()
-        self.dustThread      = DustThread(serialport,simulation=False)
+        self.dustThread      = DustThread(SERIALPORT,simulation=False)
         self.snapshotThread  = SnapshotThread(self.dustThread)
         self.fileThread      = FileThread()
         self.sendThread      = SendThread()
-        self.jsonThread      = JsonThread(tcpport,self.dustThread)
-    
+        self.jsonThread      = JsonThread(TCPPORT,self.dustThread)
+        log.debug("All threads stated")
+
+    def restart(self):
+        log.debug("Restarting threads")
+        self.close()
+        time.sleep(3)
+        self.dustThread      = DustThread(SERIALPORT,simulation=False)

```

Comment:
don't copy-paste this and  the `__init__()` code, rather implement a `start()` method